extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/filters/CEAttributeLiteralEqualsFilter.java

```diff
@@ -0,0 +1,84 @@
+package io.quarkus.funqy.runtime.bindings.knative.events.filters;
+
+import java.util.function.Predicate;
+
+import io.quarkus.funqy.knative.events.CloudEvent;
+
+public class CEAttributeLiteralEqualsFilter implements Predicate<CloudEvent> {
+
+    private String attributeName;
+    private String expectedValue;
+
+    public CEAttributeLiteralEqualsFilter(String attributeName, String expectedValue) {
+        this.attributeName = attributeName;
+        this.expectedValue = expectedValue;
+    }
+
+    @Override
+    public boolean test(CloudEvent event) {
+
+        CloudEvent<?> ceEvent = (CloudEvent<?>) event;
+        String value;
+
+        switch (attributeName) {
+            case "id":
+                value = ceEvent.id();
+                break;
+            case "source":
+                value = ceEvent.source();
+                break;
+            case "type":
+                value = ceEvent.type();
+                break;
+            case "subject":
+                value = ceEvent.subject();
+                break;
+
+            default:
+                value = ceEvent.extensions().get(attributeName);
+                break;
+        }
+        if (value == null) {
```

Comment:
Not sure this would be useful but: what if user wants to check that extension is not present? Do you think that it could be useful?

Reply:
I'd defer this to later as this filter is mainly designed to do the `equals` comparison - see class name. Later on we could expand to more filter types.